### PR TITLE
Filter Search For Admin added

### PR DESCRIPTION
**Pull request by mrgjune**
### :scroll: Description

- Describe the general shape of this PR (new feature? refactor? bug fix? one-line change?)
Added a Admin Search Bar to filter by Name, Email, Zip and Phone Number
Using Fuse.js for the filtering

- Describe why these changes are being made
Adding a way a admin can  search for someone 

>

- Describe any external libraries/dependencies added or removed in this PR

Fuse.JS

- Describe any security risks are there regarding this change

- Describe how you tested these changes
Functional Tests

--------------------------
### :clipboard: Mandatory Checklist

- [x] Example of a checked item (please remove when creating your Pull Request) 

- [x] Linked to the Github Issues being addressed using the right sidebar :arrow_right:
- [x] Have you discussed these changes with the project leader(s)?
- [x] Do all variable and function names communicate what they do?
- [x] Were all the changes commented and / or documented?
- [ x Is the PR the right size? (If the PR is too large to review, it might be good to break it up into multiple PRs.)
- [x] Does all work in progress, temporary, or debugger code have a TODO comment with links to Github issues?
- x ] *If you changed the user interface, did you add before and after screenshots to below?*

--------------------------
### :framed_picture: Screenshots and Screen Recordings
Screen Shot 2021-06-11 at 4 12 48 PM

--------------------------
### :blue_book: Glossary
- PR = Pull Request

### CODE REVIEW

File: src/containers/AdminDashboard/AdminDashboard.js [hunk 1]
```diff
@@ -339,7 +342,19 @@ const AdminDashboard = (props) => {
...
 
     const responsesMarkup = useMemo(() => {
-        return questionnaireResponses.map((response, index) => {
+        let filterQuestionnaireResponses;
+        if (searchTerm) {
+            const fuse = new Fuse(questionnaireResponses, {
+                keys: [`questionnaireResponse.${filterBy}`],
+                minMatchCharLength: 3,
+            });
+            filterQuestionnaireResponses = fuse
```
Comment (JMStudiosJoe) [inline src/containers/AdminDashboard/AdminDashboard.js:351]: what is the fuse for? searching through? we should be able to do that with raw js and not need to add on another library but I might be missing something.

**Review by JMStudiosJoe — CHANGES_REQUESTED** [review]

File: src/containers/AdminDashboard/AdminDashboard.js (hunk 1 above)
Comment (JMStudiosJoe) [reply src/containers/AdminDashboard/AdminDashboard.js:351]: if you need help gong through the logic to this feel free to ping me.

Comment (mrgjune) [reply src/containers/AdminDashboard/AdminDashboard.js:351]: if we are sure we only want a simple case, exact substring(possibly case insensitive?) matching then the matching would be easy to with js string methods. Fuse would it make it easy to do fuzzy matching (things close to the word). Let me know what you think happy to change it.

**Review by JMStudiosJoe — APPROVED** [review]
fantastic

**Pull request merged**